src/core/callouts/qgscallout.cpp

```diff
@@ -135,8 +140,24 @@ QgsCallout::DrawOrder QgsCallout::drawOrder() const
 
 void QgsCallout::render( QgsRenderContext &context, const QRectF &rect, const double angle, const QgsGeometry &anchor, QgsCalloutContext &calloutContext )
 {
-#if 0 // for debugging
   QPainter *painter = context.painter();
+  QgsScopedQPainterState painterState( painter );
```

Comment:
If you’re manually resetting the composition mode at the end of this function then you can skip this and save some cycles (or even better, make a scope painter composition mode  override class)